rest-api/test/unit_test/api_test/biobank_order_api_test.py

```diff
@@ -17,9 +19,275 @@ def setUp(self):
+    result = self.send_post(self.path, order_json)
+    full_order_json = load_biobank_order_json(self.participant.participantId,
+                                              filename='biobank_order_1.json')
+    _strip_fields(result)
+    _strip_fields(full_order_json)
+    self.assertEquals(full_order_json, result)
+
+    biobank_order_id = result['identifier'][1]['value']
+    path = self.path + '/' + biobank_order_id
+    request_data = {
+      "amendedReason": "Its all wrong",
+      "cancelledInfo": {
+        "author": {
+          "system": "https://www.pmi-ops.org/healthpro-username",
+          "value": "fred@pmi-ops.org"
+        },
+        "site": {
+          "system": "https://www.pmi-ops.org/site-id",
+          "value": "hpo-site-monroeville"
+        }
+      },
+      "status": "cancelled"
+    }
+    self.send_patch(path, request_data=request_data, headers={'If-Match': 'W/"1"'})
+
+    request_data = {
+      "amendedReason": "I didnt mean to cancel",
+      "restoredInfo": {
+        "author": {
+          "system": "https://www.pmi-ops.org/healthpro-username",
+          "value": "fred@pmi-ops.org"
+        },
+        "site": {
+          "system": "https://www.pmi-ops.org/site-id",
+          "value": "hpo-site-monroeville"
+        }
+      },
+      "status": "restored"
+    }
+
+    self.send_patch(path, request_data=request_data, headers={'If-Match': 'W/"2"'})
+    restored_order = self.send_get(path)
+    self.assertEqual(restored_order['status'], 'UNSET')
+    self.assertEqual(restored_order['restoredInfo']['author']['value'], 'fred@pmi-ops.org')
+    self.assertEqual(restored_order['restoredInfo']['site']['value'], 'hpo-site-monroeville')
+    self.assertEqual(restored_order['amendedReason'], 'I didnt mean to cancel')
+
+  def test_amending_an_order(self):
+    # pylint: disable=unused-variable
+    self.summary_dao.insert(self.participant_summary(self.participant))
+    order_json = load_biobank_order_json(self.participant.participantId,
+                                         filename='biobank_order_2.json')
+    result = self.send_post(self.path, order_json)
+    load_biobank_order_json(self.participant.participantId,
+                            filename='biobank_order_1.json')
+
+    biobank_order_id = result['identifier'][1]['value']
+    path = self.path + '/' + biobank_order_id
+    request_data = {
+      "amendedReason": "Its all better",
+      "amendedInfo": {
+        "author": {
+          "system": "https://www.pmi-ops.org/healthpro-username",
+          "value": "fred@pmi-ops.org"
+        },
+        "site": {
+          "system": "https://www.pmi-ops.org/site-id",
+          "value": "hpo-site-bannerphoenix"
+        }
+      }
+    }
+    get_order = self.send_get(path)
+    full_order = get_order.copy()
+    full_order.update(request_data)
+
+    self.assertEqual(len(full_order['samples']), 16)
+    del get_order['samples'][0]
```

Comment:
I don't get how deleting from get_order here winds up mutating full_order... I guess because get_order.copy isn't a deep copy? 

Why not delete from full_order instead?

Reply:
done